ruma-events-macros/src/gen.rs

```diff
@@ -180,7 +180,7 @@ fn populate_state_fields(content_name: Ident, fields: Vec<Field>) -> Vec<Field>
 
 /// Splits the given `event_type` string on `.` and `_` removing the `m.` then
 /// camel casing to give the `EventType` variant.
-fn to_camel_case(name: String) -> String {
+pub(crate) fn to_camel_case(name: String) -> String {
```

Comment:
This function is only ever called with `lit_str.value()` and we really should be preserving spans for it (as we do in this file [on the master branch](https://github.com/ruma/ruma-events/blob/a3f377ec39490745de667d0916edfd7bc49a4289/ruma-events-macros/src/gen.rs#L83-L84)); why not make that mandatory by taking a `LitStr` and returning an `Ident`:

```suggestion
pub(crate) fn to_camel_case(name: LitStr) -> Ident {
```

Also, since this function is not actually used in this file on the api22 branch, let's move it to `collection.rs`.